docs/cri/decryption.md

```diff
@@ -15,32 +15,39 @@ In this model encryption is tied to worker nodes. The usecase here revolves arou
+This is the default model since containerd v1.5.
 
-An example for configuring the "node" key model for container image decryption:
-
-Configure `cri` to enable decryption with "node" key model
+For containerd v1.4, you need to add the following configuration to `/etc/containerd/config.toml` manually.
 ```toml
+version = 2
+
 [plugins."io.containerd.grpc.v1.cri".image_decryption]
   key_model = "node"
-```
 
-Configure `containerd` daemon [`stream_processors`](https://github.com/containerd/containerd/blob/master/docs/stream_processors.md) to handle the
-encrypted mediatypes.
-```toml
 [stream_processors]
   [stream_processors."io.containerd.ocicrypt.decoder.v1.tar.gzip"]
     accepts = ["application/vnd.oci.image.layer.v1.tar+gzip+encrypted"]
     returns = "application/vnd.oci.image.layer.v1.tar+gzip"
-    path = "/usr/local/bin/ctd-decoder"
-    args = ["--decryption-keys-path", "/keys"]
+    path = "ctd-decoder"
+    args = ["--decryption-keys-path", "/etc/containerd/ocicrypt/keys"]
   [stream_processors."io.containerd.ocicrypt.decoder.v1.tar"]
     accepts = ["application/vnd.oci.image.layer.v1.tar+encrypted"]
     returns = "application/vnd.oci.image.layer.v1.tar"
-    path = "/usr/local/bin/ctd-decoder"
-    args = ["--decryption-keys-path", "/keys"]
+    path = "ctd-decoder"
+    args = ["--decryption-keys-path", "/etc/containerd/ocicrypt/keys"]
 ```
 
-In this example, container image decryption is set to use the "node" key model. In addition, the decryption [`stream_processors`](https://github.com/containerd/containerd/blob/master/docs/stream_processors.md) are configured as specified in [containerd/imgcrypt project](https://github.com/containerd/imgcrypt), with the additional field `--decryption-keys-path` configured to specify where decryption keys are located locally in the node.
+In this example, container image decryption is set to use the "node" key model.
+In addition, the decryption [`stream_processors`](https://github.com/containerd/containerd/blob/master/docs/stream_processors.md) are configured as specified in [containerd/imgcrypt project](https://github.com/containerd/imgcrypt), with the additional field `--decryption-keys-path` configured to specify where decryption keys are located locally in the node.
+
+After modifying this config, you need restart the `containerd` service.
 
-After modify this config, you need restart the `containerd` service.
+#### ocicrypt keyprovider protocol
+To use [ocicrypt keyprovider protocol](https://github.com/containers/ocicrypt/blob/master/docs/keyprovider.md), specicy `$OCICRYPT_KEYPROVIDER_CONFIG` as follows in `/etc/containerd/config.toml`.
+```toml
+  [stream_processors."io.containerd.ocicrypt.decoder.v1.tar.gzip"]
+    accepts = ["application/vnd.oci.image.layer.v1.tar+gzip+encrypted"]
+    returns = "application/vnd.oci.image.layer.v1.tar+gzip"
+    path = "ctd-decoder"
+    env= ["OCICRYPT_KEYPROVIDER_CONFIG=/path/to/ocicrypt.conf"]
```

Comment:
@stefanberger Can we have some default value for this?

Reply:
I would call it `/etc/ocicrypt_keyprovider.conf`. I would NOT call it `ocicrypt.conf`, because this is the default name used by an other config file that was introduced earlier: https://github.com/containers/ocicrypt/blob/master/config/pkcs11config/config.go#L35

@lumjjb  ?

Reply:
`/etc/ocicrypt_keyprovider.conf` sounds fine. We probably want to test this first with `ctd-decoder` tests to see if it will ignore the case of a file that isn't present.

Reply:
Added test : https://github.com/containerd/imgcrypt/pull/40

Reply:
If we are going to place `/etc/ocicrypt_keyprovider.conf` outside `/etc/containerd`, do we want to have `decryption-keys-path` outside `/etc/containerd` too?

Or can we have everything under `/etc/containerd/ocicrypt`?

Reply:
I am fine with putting it in `/etc/containerd/ocicrypt/ocicrypt_keyprovider.conf`, i like the idea, would probably prevent some conflicts with other tools. 

Btw, would the `ctd-decoder` also be packaged with the installation?

Reply:
Updated PR to add `env =  ["OCICRYPT_KEYPROVIDER_CONFIG=/etc/containerd/ocicrypt/ocicrypt_keyprovider.conf"]`

> Btw, would the ctd-decoder also be packaged with the installation?

SGTM, but out of the scope of this PR.

Reply:
Opened https://github.com/containerd/containerd/issues/5265 for discussing inclusion of `ctd-decoder`